pythran/toolchain.py

```diff
@@ -343,18 +349,24 @@ def compile_pythranfile(file_path, output_file=None, module_name=None,
     """
     if not output_file:
         # derive module name from input file name
-        _, basename = os.path.split(file_path)
+        dirname, basename = os.path.split(file_path)
         module_name = module_name or os.path.splitext(basename)[0]
 
     else:
         # derive module name from destination output_file name
-        _, basename = os.path.split(output_file)
+        dirname, basename = os.path.split(output_file)
         module_name = module_name or os.path.splitext(basename)[0]
 
     # Add compiled module path to search for imported modules
     sys.path.append(os.path.dirname(file_path))
 
-    output_file = compile_pythrancode(module_name, open(file_path).read(),
+    specs = None
+    spec_file = os.path.join(dirname, module_name + ".pythran")
+    if os.path.isfile(spec_file):
+        from pythran.spec import spec_parser
+        specs = spec_parser(spec_file, False)
```

Comment:
here, instead of the bool, you would use an `aux_spec_parser` that crates the deriving object etc